src/blockchain/deposit_strategy/deposit_transaction_sender.py

```diff
@@ -28,50 +42,115 @@ def _prepare_signs_for_deposit(quorum: list[DepositMessage]) -> tuple[tuple[str,
 
     def prepare_and_send(
         self,
+        module_id: int,
         quorum: list[DepositMessage],
         with_flashbots: bool,
-        is_mellow: bool,
     ) -> bool:
-        tx = self._prepare_mellow_tx(quorum) if is_mellow else self._prepare_general_tx(quorum)
-        return self._send_transaction(tx, with_flashbots)
+        if self._sender_checks(module_id):
+            tx = self._prepare_tx(quorum)
+            if tx and self._send_transaction(tx, with_flashbots):
+                return True
+
+        if self._next_sender is not None:
+            return self._next_sender.prepare_and_send(module_id, quorum, with_flashbots)
+
+        return False
 
-    def _prepare_mellow_tx(self, quorum: list[DepositMessage]) -> ContractFunction:
+    def _prepare_tx(self, quorum: list[DepositMessage]) -> Optional[ContractFunction]:
         block_number = quorum[0]['blockNumber']
         block_hash = Hash32(bytes.fromhex(quorum[0]['blockHash'][2:]))
         deposit_root = Hash32(bytes.fromhex(quorum[0]['depositRoot'][2:]))
+        staking_module_id = quorum[0]['stakingModuleId']
         staking_module_nonce = quorum[0]['nonce']
         payload = b''
         guardian_signs = self._prepare_signs_for_deposit(quorum)
-        return self._w3.lido.simple_dvt_staking_strategy.convert_and_deposit(
+        return self._w3.lido.deposit_security_module.deposit_buffered_ether(
             block_number,
             block_hash,
             deposit_root,
+            staking_module_id,
             staking_module_nonce,
             payload,
             guardian_signs,
         )
 
-    def _prepare_general_tx(self, quorum: list[DepositMessage]):
+    def _send_transaction(
+        self,
+        tx: ContractFunction,
+        flashbots_works: bool
+    ) -> bool:
+        return self._w3.transaction.check(tx) and self._w3.transaction.send(tx, flashbots_works, self._TIMEOUT_IN_BLOCKS)
+
+    def _sender_checks(self, module_id: int) -> bool:
```

Comment:
Not sure `Sender` should know anything about strategies and etc.
Looks like `Sender` should prepare and send transaction

Reply:
Removed